xdsl/transforms/lower_riscv_func.py

```diff
@@ -64,15 +64,15 @@ def match_and_rewrite(self, op: riscv_func.SyscallOp, rewriter: PatternRewriter)
 class LowerRISCVFuncOp(RewritePattern):
     @op_type_rewrite_pattern
     def match_and_rewrite(self, op: riscv_func.FuncOp, rewriter: PatternRewriter):
-        body = op.func_body.block
-        first_op = body.first_op
+        first_block = op.func_body.blocks[0]
+        first_op = first_block.first_op
         assert first_op is not None
-        while len(body.args):
+        while len(first_block.args):
```

Comment:
I think it would be more readable to use for loop instead of `while` with extra variables.

Run and tested:
```python
        for index, arg in enumerate(first_block.args):
            get_reg_op = riscv.GetRegisterOp(riscv.IntRegisterType(f"a{index}"))
            rewriter.insert_op_before(get_reg_op, first_op)
            arg.replace_by(get_reg_op.res)
            rewriter.erase_block_argument(arg)
```

Reply:
it is not allowed to iterate over a collection that is being mutated in Python

Reply:
You could clone the list before iterating maybe? `enumerate(tuple(first_block.args))`?

Reply:
yeah that's a good point, although I'm not a big fan of the temporary allocation... A better way might be to do two loops